### ISSUE TITLE
[v1.1.2] Local resolv.conf config being pointed to ctrld without --iface setting

### ISSUE THREAD

**Issue by chr-lei** [Control-D-Inc/ctrld#30]
ctrld version: v1.1.2 / arm64
OS: Debian 11 (bullseye)

When starting ctrld as a service via `ctrld service start` the service is updating the resolv.conf file on the system to use the listener's IP address as the nameserver, even if `--iface=auto` or a specific interface is not specified. This causes issues with domains that are configured in a Policy to use the local resolver; an infinite loop is created which seems to crash `ctrld` eventually.

If the daemon is started via `ctrld run` with the same configuration file argument, the resolv.conf file is not modified which is the expected behavior.

I did confirm that the systemd service for `ctrld` does not include `--iface=auto` when it's created via `ctrld service start`, so I am not sure why it's still messing with the DNS servers on the interface when it's run as a service but not when it's run via `ctrld run`.

My original /etc/resolv.conf file is a single line:
`nameserver 172.27.0.1`

I install & start the service via:
`./ctrld service start --config=/opt/ctrld/ctrld.toml`

which gives the following debug output (note the updates to the `eth0` iface)
```
loading config file from: /opt/ctrld/ctrld.toml
Mar 17 09:17:09.352 DBG Performing self-check
Mar 17 09:17:10.392 DBG self-check against "verify.controld.com" succeeded
Mar 17 09:17:10.392 INF Service started
Mar 17 09:17:10.440 DBG setup NetworkManager done
Mar 17 09:17:10.441 DBG setting DNS for interface iface=eth0
Mar 17 09:17:10.446 DBG dns: [rc=unknown ret=direct]
Mar 17 09:17:10.446 DBG dns: using "direct" mode
Mar 17 09:17:10.515 DBG setting DNS successfully iface=eth0
```
and now /etc/resolv.conf contains:
```
# resolv.conf(5) file generated by ctrld
# DO NOT EDIT THIS FILE BY HAND -- CHANGES WILL BE OVERWRITTEN

nameserver 172.27.0.10
```

Confirming via `systemctl` that the service isn't using --iface=auto or --iface=eth0 as an argument:
```
# systemctl status ctrld.service
● ctrld.service
     Loaded: loaded (/etc/systemd/system/ctrld.service; enabled; vendor preset: enabled)
     Active: active (running) since Fri 2023-03-17 09:17:09 EDT; 2min 28s ago
   Main PID: 186156 (ctrld)
      Tasks: 10 (limit: 1830)
        CPU: 488ms
     CGroup: /system.slice/ctrld.service
             └─186156 /opt/ctrld/ctrld run --config=/opt/ctrld/ctrld.toml --homedir=/root
```

If I run the `ctrld run` command shown in the systemctl output from a regular shell prompt, ctrld does not seem to mess with DNS on any interfaces:
```
# /opt/ctrld/ctrld run --config=/opt/ctrld/ctrld.toml --homedir=/root
loading config file from: /opt/ctrld/ctrld.toml
starting ctrld...
Mar 17 09:22:29.764 DBG Resolving "dns.controld.com" using bootstrap DNS ["76.76.2.0:53" "172.27.0.1:53"]
Mar 17 09:22:29.789 DBG Bootstrap IPs: [2606:1a40::22 76.76.2.22]
Mar 17 09:22:29.790 INF Setting bootstrap IP for upstream.0 bootstrap_ip=76.76.2.22
Mar 17 09:22:29.790 DBG sending doh request to: 76.76.2.22:443
Mar 17 09:22:30.069 INF Starting DNS server on listener.0: 172.27.0.10:53
Mar 17 09:22:30.322 DBG [c513f1] 172.21.0.116:51444 -> listener.0: 172.27.0.10:53: received query: A gsa.apple.com
Mar 17 09:22:30.322 INF [c513f1] no policy, no network, no rule -> [upstream.0]
Mar 17 09:22:30.322 DBG [c513f1] sending query to upstream.0: 
```
and /etc/resolv.conf remains intact:
```
# cat /etc/resolv.conf
nameserver 172.27.0.1
```

My `ctrld.toml` file is pretty simple (resolver ID is redacted but otherwise this is the full file):
```
[service]
 cache_enable = "true"
 cache_size = 4096
 log_level = "debug"
 log_path = "/var/log/ctrld.log"

[listener]
  [listener.0]
    ip = '172.27.0.10'
    port = 53

[listener.0.policy]
  name = "My Policy"
  rules = [
   {"*.homenet.crlonline.net" = []},
   {"*.corpnet.susquehannaitc.com" = []},
   {"*.20.172.in-addr.arpa" = []},
   {"*.21.172.in-addr.arpa" = []},
   {"*.22.172.in-addr.arpa" = []}
  ]

[network]
  [network.0]
    name = 'Network 0'
    cidrs = ['0.0.0.0/0']

[upstream]
  [upstream.0]
    type = 'doh'
    endpoint = 'https://dns.controld.com/resolverID'
    timeout = 5000
```

Hopefully I am not missing something obvious here. Let me know if there's more info I can provide or other testing I can do.

**Comment by cuonglm** [comment]
@chr-lei You can do `./ctrld service start --config=/opt/ctrld/ctrld.toml --iface=""` to workaround this bug.

**Comment by yegors** [comment]
Resolved in v1.1.3
